src/core/layout/qgslayoutatlas.cpp

```diff
@@ -439,18 +439,18 @@ QString QgsLayoutAtlas::currentFilename() const
   return mCurrentFilename;
 }
 
-QgsExpressionContext QgsLayoutAtlas::createExpressionContext()
+QgsExpressionContext QgsLayoutAtlas::createExpressionContext() const
 {
   QgsExpressionContext expressionContext;
   expressionContext << QgsExpressionContextUtils::globalScope();
   if ( mLayout )
     expressionContext << QgsExpressionContextUtils::projectScope( mLayout->project() )
                       << QgsExpressionContextUtils::layoutScope( mLayout );
 
-  expressionContext.appendScope( QgsExpressionContextUtils::atlasScope( this ) );
+  expressionContext.appendScope( QgsExpressionContextUtils::atlasScope( const_cast< QgsLayoutAtlas * >( this ) ) );
```

Comment:
Any reason not to const-change the signature `QgsExpressionContextUtils::atlasScope`?

Reply:
I tried originally, but there's non const getters used in atlasScope and the changes started snowballing...

Reply:
Actually it wasn't that bad in the end!